### PR TITLE
Fix input focusing issues

### PR DESCRIPTION
**Pull request by mikedemarais**
- Consolidates reusable Wallet / Contact related expanded state components
Fixes RAI-742 https://linear.app/rainbow/issue/RAI-742/remaining-input-focusing-issues
Fixes RAI-670 https://linear.app/rainbow/issue/RAI-670/spacing-on-the-import-seed-phrase-sheet-and-send-sheet-is-inconsistent
Fixes RAI-746 https://linear.app/rainbow/issue/RAI-746/importing-doesnt-work-on-iphone-8-ios-124

### CODE REVIEW

**Comment by christianbaroni** [comment]
not urgent but #2 from here still an issue: https://linear.app/rainbow/issue/RAI-742/remaining-input-focusing-issues

**Comment by christianbaroni** [comment]
positioning of this modal is a little off

before (left) and after changes:
Screen Shot 2020-07-13 at 12 53 05 PM

**Comment by christianbaroni** [comment]
also iOS 12.4 iPhone 8, import sheet input loses focus when opening

might also happen with send sheet

https://www.dropbox.com/s/9u2cslbk5tflxvi/Screen%20Recording%202020-07-13%20at%2012.53.33%20PM.mov?dl=0

**Comment by mikedemarais** [comment]
> also iOS 12.4 iPhone 8, import sheet input loses focus when opening
> 
> might also happen with send sheet
> 
> https://www.dropbox.com/s/9u2cslbk5tflxvi/Screen%20Recording%202020-07-13%20at%2012.53.33%20PM.mov?dl=0

yup! been talking with @osdnk about this -- the solution with solve `#5` as well

**Comment by mikedemarais** [comment]
> positioning of this modal is a little off
> 
> before (left) and after changes:
> Screen Shot 2020-07-13 at 12 53 05 PM

i think mine is closer to the spec, previous was implementation of old spec i thinkkk

**Comment by christianbaroni** [comment]
> > positioning of this modal is a little off
> > before (left) and after changes:
> > Screen Shot 2020-07-13 at 12 53 05 PM
> 
> i think mine is closer to the spec, previous was implementation of old spec i thinkkk

ahh I'm only talking about the vertical alignment of the modal, what's inside the modal looks good

File: src/components/contacts/ContactAvatar.js [hunk 1]
```diff
@@ -1,44 +1,48 @@
...
-      shadows: [
-        [0, 3, 5, colors.dark, 0.14],
-        [0, 6, 10, colors.avatarColor[color] || color, 0.2],
-      ],
-      textSize: 'large',
-    },
-  };
-  const { dimensions, shadows, textSize } = sizeConfigs[size];
+const ContactAvatar = ({ color, size = 'medium', value, ...props }) => {
+  const { dimensions, shadow, textSize } = useMemo(() => sizeConfigs[size], [
```
Comment (brunobar79) [inline src/components/contacts/ContactAvatar.js:38]: I think reading a key from an object directly is probably cheaper than useMemo

File: src/components/modal/LoadingOverlay.js [hunk 2]
```diff
@@ -8,18 +7,15 @@ import TouchableBackdrop from '../TouchableBackdrop';
...
-      align-self: center;
-      flex: 1;
-    `
-    : ''}
+const Container = styled(Centered).attrs({
+  flex: Platform.OS === 'android' ? 1 : undefined,
+  self: Platform.OS === 'android' ? 'center' : undefined,
+})`
   ${position.size('100%')};
+  position: absolute;
```
Comment (brunobar79) [inline src/components/modal/LoadingOverlay.js:17]: Do we need this?

File: src/screens/ChangeWalletSheet.js [hunk 3]
```diff
@@ -362,7 +358,7 @@ const ChangeWalletSheet = () => {
           <Divider color={colors.rowDividerExtraLight} inset={[0, 15]} />
         )}
       </Column>
-      <EditButton onPress={toggleEditMode}>
+      <EditButton onPress={() => setEditMode(e => !e)}>
```
Comment (brunobar79) [inline src/screens/ChangeWalletSheet.js:361]: Don't you wanna keep the useCallback around here?

Comment (osdnk) [reply src/screens/ChangeWalletSheet.js:361]: I think it's not worth memoizing

File: src/components/expanded-state/profile/ProfileAvatarButton.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React, { useCallback } from 'react';
+import { ButtonPressAnimation } from '../../animations';
+import { ContactAvatar } from '../../contacts';
+import { colors } from '@rainbow-me/styles';
+
+const ProfileAvatarButton = ({ color, marginBottom = 15, setColor, value }) => {
+  const handleChangeColor = useCallback(() => {
+    if (setColor) {
+      setColor(prevColor => {
```
Comment (jinchung) [inline src/components/expanded-state/profile/ProfileAvatarButton.js:9]: `setColor(prevColor => (prevColor + 1) % colors.avatarColor.length);`

**Pull request merged**